Core/include/Acts/Vertexing/AdaptiveMultiVertexFitter.hpp

```diff
@@ -98,6 +104,23 @@ class AdaptiveMultiVertexFitter {
         }
       }
     }
+
+    Result<void> removeVertexFromCollection(Vertex<InputTrack_t>& vtxToRemove) {
+      auto it = std::find(vertexCollection.begin(), vertexCollection.end(),
+                          &vtxToRemove);
+      // Check if the value was found before erasing
+      if (it == vertexCollection.end()) {
+        ACTS_ERROR("vtxToRemove is not part of vertexCollection.");
+        return VertexingError::InvalidMemoryAccess;
```

Comment:
I guess this could just reuse `ElementNotFound` which already exist in the error enum?